react/components/SKUSelector/index.js

```diff
@@ -1,110 +1,150 @@
 import PropTypes from 'prop-types'
-import React, { useState, useEffect, useCallback, useMemo, memo } from 'react'
+import React, { useState, useEffect, useMemo, memo, useCallback } from 'react'
 import { useRuntime } from 'vtex.render-runtime'
+import { filter, head, isEmpty } from 'ramda'
 
 import SKUSelector from './components/SKUSelector'
 import { skuShape } from './utils/proptypes'
 import {
-  groupItemsByVariation,
-  getMainVariationName,
-  parseSku,
+  parseSku, isColor, uniqueOptionToSelect, findItemWithSelectedVariations,
 } from './utils'
 
+
+const buildEmptySelectedVariation = (variations) => {
+  const variationNames = Object.keys(variations)
+  const result = {}
+  for (const varName of variationNames) {
+    result[varName] = null
+  }
+  return result
+}
+
+/** receives an item and the variations object, returns the selected variations for that item */
+const selectedVariationFromItem = (item, variations) => {
+  const variationNames = Object.keys(variations)
+  const result = {}
+  for (const varName of variationNames) {
```

Comment:
⛏ please use `variationName` here (no need for shortening + it can be confused with "variable name")